### PR TITLE
fix: Clean up imports so that code runs with core dependencies. Updat…

### PR DESCRIPTION
**Pull request by mseeger**
…e tests to enforce this

Need to be more careful with imports in the future!

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mseeger** [comment]
There are still some `end-to-end-tests` failing

**Comment by mseeger** [comment]
OK, `end-to-end-tests` are working now

File: .github/workflows/run-syne-tune.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
           python -m pip install --upgrade pip
       - name: Install Syne Tune with all dependencies
```
Comment (wesk) [inline .github/workflows/run-syne-tune.yml:56]: Nit
```suggestion
      - name: Install Syne Tune with core dependencies only
```

**Review by wesk — APPROVED** [review]
Nice change, and the stricter tests will help prevent this from happening again.

I verified that with this change it's now possible to run `from syne_tune import Tuner` with just core dependencies.

**Review by wesk — APPROVED** [review]

**Pull request merged**